### PR TITLE
Introduce partial typing for player statistics fetching

### PR DESCRIPTION
**Pull request by hburn7**
Attempts to partially introduce typing for the `AreaChart` type (which is responsible for displaying the line chart on player profiles). I was getting stuck on some of the anonymous types but at least things are a *little* better now.

Also:

- Renames some variables to make more sense
- Attempts to clean up some of the old code in general

### CODE REVIEW

File: app/actions.ts [hunk 1]
```diff
@@ -330,57 +330,29 @@ export async function fetchUserPageTitle(player: string | number) {
   return null;
 }
 
-export async function fetchUserPage(player: string | number, params) {
-  const session = await getSessionData();
-
+export async function fetchPlayerStats(player: string | number, params) {
```
Comment (myssto) [inline app/actions.ts:333]: Move this function to `/app/actions/players.ts`

File: app/players/[id]/page.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ export default async function page({
   searchParams: URLSearchParams;
```
Comment (myssto) [inline app/players/[id]/page.tsx:36]: `searchParams` is not of type `URLSearchParams`. See [next docs](https://nextjs.org/docs/app/api-reference/file-conventions/page#searchparams-optional)

**Review by myssto — CHANGES_REQUESTED** [review]

File: app/players/[id]/page.tsx (hunk 2 above)
Comment (hburn7) [reply app/players/[id]/page.tsx:36]: This one I will leave to you or Aki, this is used in 5 different places.

Comment (AkinariHex) [reply app/players/[id]/page.tsx:36]: @myssto for now I would like to leave as `URLSearchParams` and then changing better another time

**Review by AkinariHex — APPROVED** [review]

File: app/players/[id]/page.tsx (hunk 2 above)
Comment (myssto) [reply app/players/[id]/page.tsx:36]: sure

**Review by myssto — APPROVED** [review]

**Pull request merged**